pkg/controller/statefulset/stateful_set_utils.go

```diff
@@ -316,8 +316,13 @@ func newRevision(set *apps.StatefulSet, revision int64, collisionCount *int32) (
 	if err != nil {
 		return nil, err
 	}
+	podLabels := set.Spec.Template.Labels
+	if podLabels == nil {
```

Comment:
podLabels should never be nil

- The statefulset selector is required to be [non-empty](https://github.com/kubernetes/kubernetes/blob/05944b1d2ca7f60b09762a330425108f48f6b603/pkg/apis/apps/validation/validation.go#L106-L113)
- [The podtemplate labels are required to match the selector](https://github.com/kubernetes/kubernetes/blob/05944b1d2ca7f60b09762a330425108f48f6b603/pkg/apis/apps/validation/validation.go#L50-L53).

Reply:
I had this doubt, did the check anyway to be on safer side. Removed the check now. Please see the recent commit.